onyo/lib/tests/test_git.py

```diff
@@ -0,0 +1,264 @@
+from pathlib import Path
+
+import pytest
+from onyo import OnyoRepo, OnyoInvalidRepoError
+from onyo.lib.git import GitRepo
+
+
+def test_GitRepo_instantiation_existing(repo: OnyoRepo) -> None:
+    """
+    The GitRepo class must instantiate correctly for paths to existing
+    repositories.
+    """
+    new_repo = GitRepo(repo.git.root)
+    assert new_repo.root.samefile(repo.git.root)
```

Comment:
General remark on that (not only this particular test): Using the `repo` fixture here, seems a bit off. `GitRepo` doesn't require an `OnyoRepo`. Ever.

Also: If the fixture was able to instantiate an `OnyoRepo` that delivers a `.git` which you can use to pass `repo.git.root`, then there already was a successful instantiation of a `GitRepo` on that path. Hence, this test does nothing.

Reply:
What you'd want would be a temporary directory, run `git init` on it via `subprocess` and then try and instantiate this object.

Reply:
> Hence, this test does nothing.

That is not fully true. This test-function verifies for a path/repository that already exists as git, that calling `GitRepo(root)` instantiates the GitRepo object (not the git repository itself) and sets the root for the new object correctly.

> What you'd want would be a temporary directory, run `git init` on it via `subprocess` and then try and instantiate this object.

I agree, and will add such a test.

Reply:
Add-on explenation: I do not add "such a test", but change all OnyoRepos to temp_paths and initialize them with git init and subprocess like you described.

Reply:
> This test-function verifies for a path/repository that already exists as git, that calling GitRepo(root) instantiates the GitRepo object (not the git repository itself) and sets the root for the new object correctly.

Well not really. This needed to succeed in the fixture before the test function was even executed. It tested its own prerequirement.